src/autolabel/models/google.py

```diff
@@ -0,0 +1,180 @@
+from time import time
+from typing import List, Optional
+
+from langchain.schema import Generation, HumanMessage, LLMResult
+from tenacity import before_sleep_log, retry, stop_after_attempt, wait_exponential
+
+from autolabel.cache import BaseCache
+from autolabel.configs import AutolabelConfig
+from autolabel.models import BaseModel
+from autolabel.schema import ErrorType, LabelingError, RefuelLLMResult
+
+logger = logging.getLogger(__name__)
+
+
+class GoogleLLM(BaseModel):
+    SEP_REPLACEMENT_TOKEN = "@@"
+    CHAT_ENGINE_MODELS = ["gemini-pro"]
+
+    DEFAULT_MODEL = "gemini-pro"
+
+    # Reference: https://ai.google.dev/pricing
+    COST_PER_CHARACTER = {
+        "gemini-pro": 0.000125 / 1000,
+    }
+
+    @cached_property
+    def _engine(self) -> str:
+        if self.model_name is not None and self.model_name in self.CHAT_ENGINE_MODELS:
+            return "chat"
+        else:
+            return "completion"
+
+    def __init__(
+        self,
+        config: AutolabelConfig,
+        cache: BaseCache = None,
+    ) -> None:
+        try:
+            import tiktoken
+            from langchain_google_genai import (
+                ChatGoogleGenerativeAI,
+                HarmBlockThreshold,
+                HarmCategory,
+            )
+        except ImportError:
+            raise ImportError(
+                "tiktoken and langchain_google_genai. Please install it with the following command: pip install 'refuel-autolabel[google]'"
+            )
+
+        self.DEFAULT_PARAMS = {
+            "temperature": 0.0,
+            "topK": 3,
+            "safety_settings": {
+                HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT: HarmBlockThreshold.BLOCK_NONE,
+                HarmCategory.HARM_CATEGORY_HATE_SPEECH: HarmBlockThreshold.BLOCK_NONE,
+                HarmCategory.HARM_CATEGORY_HARASSMENT: HarmBlockThreshold.BLOCK_NONE,
+                HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: HarmBlockThreshold.BLOCK_NONE,
+            },
+        }
+
+        super().__init__(config, cache)
+        # populate model name
+        self.model_name = config.model_name() or self.DEFAULT_MODEL
+        # populate model params and initialize the LLM
+        model_params = config.model_params()
+        self.model_params = {
+            **model_params,
+            **self.DEFAULT_PARAMS,
+        }
+        if self._engine == "chat":
+            self.llm = ChatGoogleGenerativeAI(
+                model=self.model_name, **self.model_params
+            )
+        else:
+            self.llm = ChatGoogleGenerativeAI(
+                model=self.model_name, **self.model_params
+            )
+
+        self.tiktoken = tiktoken
+
+    @retry(
+        reraise=True,
+        stop=stop_after_attempt(5),
+        wait=wait_exponential(multiplier=1, min=2, max=10),
+        before_sleep=before_sleep_log(logger, logging.WARNING),
+    )
+    def _label_with_retry(self, prompts: List[str]) -> LLMResult:
```

Comment:
this is needed for refuelLLM which does not do retries by default. Langchain does retries by default for openai. Isn't the same done for google as well?